### DISCUSSION TITLE
INL HPC MOOSE app compile issue

### DISCUSSION THREAD

**Discussion by ver2king** [idaholab/moose#28917] [Q&A General]
### Bug Description

I am an external user with access to INL HPC. I have my own MOOSE app which compiles fine in my local machine. However when I tried to access INL HPC and compiled my app, the following error message (image attached).

![image](https://github.com/user-attachments/assets/a2ea62aa-4663-4fd3-a648-3b19e0666206)

I tried to re-build moose itself with clusters as sawtooth1, sawtooth2, however the same error message occured. 

### Steps to Reproduce

The procedure I performed was as below.

1. Access INL HPC via RSA 
2. ssh sawtooth1
3. module load use.moose moose-dev-openmpi/28a1964
4. cd ~/cluster_name/projects
5. Clone my MOOSE app from Github via SSH
6. moose-dev-shell
7. make -j8

### Impact

As discussed above in the description and steps to re-produce. Note: I was an intern at INL during Summer 2024, and later I applied for an external account with a different username. Data form my intern account was transferred to this new external account (I'm not sure whether this is relevant, but I think it may give a context).

**Comment by ver2king** [comment]
May anyone look into this issue for me? I have not figured out a solution by far.

**Reply by milljm** [reply]
Converting to a Discussion, as a solution will not involve any changes to actual MOOSE code.


It looks like you need to clean your build of MOOSE. It is the only explanation when running `make` in your application directory, and to be instructed on linking to a file(s) no longer present.

Lets start with cleaning moose, since you do not mention starting with a clean MOOSE clone (but you do mention starting with a clean application clone. This was the clue that told me you may have an older stale build of MOOSE.)
```bash
cd ~/cluster_name/projects/moose
git clean -xfd; git submodule foreach --recursive git clean -xfd
```

Do the same in your application, since you tried to build, and it failed. _e.g now there are libraries that are tainted with information on linking to non-existing files in your application, just like MOOSE_:


> [!CAUTION]
> _Perform a `git add` on any untracked files you may have, to save your work. As the following command WILL DELETE YOUR WORK IF YOU DO NOT:_

Once you have saved your work, perform the clean command:
```bash
cd ~/cluster_name/projects/maxwell_potential
git clean -xfd
```
Now in a clean state, you may attempt to load up the appropriate moose-dev module, and run `make` again!
